### PR TITLE
Added tests for io.protostuff.ProtobufOutput

### PR DESCRIPTION
**Pull request by freddytuxworth**
Hi,

I've analysed your code base and noticed that there are no unit tests for the `io.protostuff.ProtostuffOutput` class. I've written some tests for this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/products).

Hopefully these tests should help you detect any regressions caused by future code changes. If you'd like any more tests written for other classes, just let me know and I'd be happy to help.

### CODE REVIEW

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+import java.io.IOException;
+
+public class ProtostuffOutputTest {
+
+    @Test
+    public void testEncodeZigZag32() {
+        Assert.assertEquals(0x00000000, ProtobufOutput.encodeZigZag32(0x00000000));
+        Assert.assertEquals(0x02020202, ProtobufOutput.encodeZigZag32(0x01010101));
+        Assert.assertEquals(0x000001FF, ProtobufOutput.encodeZigZag32(0xFFFFFF00));
+        Assert.assertEquals(0x00000001, ProtobufOutput.encodeZigZag32(0xFFFFFFFF));
```
Comment (sudo-jaa) [inline protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: I know you're not a fan of edge cases, but there is nothing in this method to check that the input int is actually a signed 32 bit int. We can pass an unsigned 32 bit int to this function with no complaint, as well as just a primitive int. Would you consider it worthwhile to add in assertions for this method's behaviour when unexpected data is passed in?

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    }
+
+    @Test
+    public void testEncodeZigZag64() {
+        Assert.assertEquals(0L, ProtobufOutput.encodeZigZag64(0L));
```
Comment (sudo-jaa) [inline protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: Can we try any other inputs?

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
+package io.protostuff;
+
+import org.junit.Assert;
```
Comment (sudo-jaa) [inline protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: does the style guide for this repo prevent importing `Assert.*;` to collapse down these assertion statements?

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            new byte[]{-89, 38, 38},
+            ProtobufOutput.getTagAndRawVarInt64Bytes(4903, 38L));
+
+        Assert.assertArrayEquals(
+            new byte[]{0, 0},
+            ProtobufOutput.getTagAndRawVarInt64Bytes(0, 0L));
+    }
+
+    @Test
+    public void testWriteRawVarInt32() throws IOException {
```
Comment (James-Adam) [inline protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: Do you need to keep the throw?

**Review by sudo-jaa — COMMENTED** [review]
A couple of comments made, nothing major. Pretty much looks good to me!

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java (hunk 3 above)
Comment (freddytuxworth) [reply protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: This isn't done elsewhere in the repo so I think I'll leave it

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java (hunk 1 above)
Comment (freddytuxworth) [reply protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: I'm not sure I understand, sorry. Could you give an example of what you mean by unexpected data?

File: protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java (hunk 2 above)
Comment (freddytuxworth) [reply protostuff-core/src/test/java/io/protostuff/ProtostuffOutputTest.java]: Whoops yes good point

**Review by sudo-jaa — APPROVED** [review]

**Review by paulkennethkent — COMMENTED** [review]
Only the filename and test class need to be changed, The rest looks good to me.
